compiler/rustc_infer/src/infer/context.rs

```diff
@@ -8,7 +8,7 @@ use rustc_span::{DUMMY_SP, ErrorGuaranteed, Span};
 
 use super::{BoundRegionConversionTime, InferCtxt, RegionVariableOrigin, SubregionOrigin};
 
-impl<'tcx> rustc_type_ir::InferCtxtLike for InferCtxt<'tcx> {
+impl<'tcx> ty::InferCtxtLike for InferCtxt<'tcx> {
```

Comment:
Revert just this to `rustc_type_ir::InferCtxtLike` -- I prefer that we name anything like `InferCtxtLike` via its actual path and not its reexport. This will need to continue using the crate in `Cargo.toml`, which is fine.

Reply:
Sure, no problem